File: leetcode/Algorithm/#938-easy.java
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        return traverse(root, low, high);
    }

    public int traverse(TreeNode node, int lo, int hi) {
        if (node == null) return 0;

        int val = node.val;
        if (node.val > hi || node.val < lo) val = 0;
        
        return val + traverse(node.left, lo, hi) + traverse(node.right, lo, hi);
    }
}

File: leetcode/Algorithm/#938-easy.kt
class Solution {
    fun rangeSumBST(root: TreeNode?, low: Int, high: Int): Int {
        root ?: return 0

        var v = 0
        if (root.`val` in low..high) {
            v = root.`val`
        }

        return rangeSumBST(root.left, low, high) + rangeSumBST(root.right, low, high) + v
    }
}